lib/mcp/app/resource.rb

```diff
@@ -69,17 +69,17 @@ def list_resources(cursor: nil, page_size: nil)
 
         if page_size.nil?
           paginated = values[start_index..]
-          next_cursor = nil
+          has_next = false
         else
           paginated = values[start_index, page_size]
           has_next = start_index + page_size < values.length
-          next_cursor = has_next ? (start_index + page_size).to_s : nil
         end
 
-        {
-          resources: paginated.map { |r| format_resource(r) },
-          nextCursor: next_cursor
+        result = {
+          resources: paginated.map { |r| format_resource(r) }
         }
+        result[:nextCursor] = (start_index + page_size).to_s if has_next
+        result
```

Comment:
the simplest possible change would have been to just call `.compact` on the resulting hash - which will do what you needs - remove the nil values

Reply:
+1

For exmaple, 

```ruby
% irb
irb(main):001> result = { resources: [], nextCursor: nil }
=> {resources: [], nextCursor: nil}
irb(main):002> result.compact
=> {resources: []}
```

Reply:
Your review was very helpful. Thanks to your suggestions, the code is much cleaner and simpler.